tests/target/issue-2321/trailing_comma_always_and_brace_style_always_next_line.rs

```diff
@@ -0,0 +1,10 @@
+// rustfmt-brace_style: AlwaysNextLine
+// rustfmt-trailing_comma: Always
+
+fn lorem<S, T,>(lorem: S, ipsum: T,)
+where
+    S: Add + Sub,
+    T: Mul + Div,
+{
+    // body
+}
```

Comment:
Where clauses can appear in other contexts besides function definitions. Can we also add some examples with enum / struct definitions, and impl blocks to highlight how `brace_style` interacts with `trailing_comma` for those other language constructs?

Also including some tests with a single where clause bound (e.g. just `S: Add + Sub`) would also be helpful to highlight the difference between how a list of bounds VS a single bound is formatted.